content/200-concepts/100-components/01-prisma-schema/06-relations/200-one-to-many-relations.mdx

```diff
@@ -0,0 +1,238 @@
+
+This 1-n-relation expresses the following:
+
+- "a user can have zero or more posts"
+- "a post must always have an author"
+
+</TopBlock>
+
+## Multi-field relations in relational databases
+
+In **relational databases only**, you can also define this relation using [multi-field IDs](../../../../reference/api-reference/prisma-schema-reference#id-1):
+
+<TabbedContent tabs={[<FileWithIcon text="Relational databases only" icon="database"/>]}>
+<tab>
+
+```prisma
+model User {
+  firstName String
+  lastName  String
+  post      Post[]
+
+  @@id([firstName, lastName])
+}
+
+model Post {
+  id               Int      @id @default(autoincrement())
+  author           User     @relation(fields: [authorFirstName, authorLastName], references: [firstName, lastName])
+  authorFirstName  String   // relation scalar field (used in the `@relation` attribute above)
+  authorLastName   String   // relation scalar field (used in the `@relation` attribute above)
+}
+```
+
+</tab>
+<tab>
+</tab>
+</TabbedContent>
+
+
+## 1-n relations in the database
+
+### Relational databases
+
+The following example demonstrates how to create a 1-n relation in SQL:
+
+```sql
+CREATE TABLE "User" (
+    id SERIAL PRIMARY KEY
+);
+CREATE TABLE "Post" (
+    id SERIAL PRIMARY KEY,
+    "authorId" integer NOT NULL,
+    FOREIGN KEY ("authorId") REFERENCES "User"(id)
+);
+```
+
+Since there's no `UNIQUE` constraint on the `authorId` column (the foreign key), you can create **multiple `Post` records that point to the same `User` record**. This makes the relation a one-to-many rather than a one-to-one.
+
+The following example demonstrates how to create a 1-n relation in SQL using a composite key (`firstName` and `lastName`):
+
+```sql
+CREATE TABLE "User" (
+    firstName TEXT,
+    lastName TEXT,
+    PRIMARY KEY ("firstName","lastName")
+);
+CREATE TABLE "Post" (
+    id SERIAL PRIMARY KEY,
+    "authorFirstName" TEXT NOT NULL,
+    "authorLastName" TEXT NOT NULL,
+    FOREIGN KEY ("authorFirstName", "authorLastName") REFERENCES "User"("firstName", "lastName")
+);
+```
+
+#### Comparing one-to-one and one-to-many relations
+
+In relational databases, the main difference between a 1-1 and a 1-n-relation is that in a 1-1-relation the foreign key must have a `UNIQUE` constraint defined on it.
+
+### MongoDB
+
+For MongoDB, Prisma currently uses a [normalized data model design](https://docs.mongodb.com/manual/core/data-model-design/), which means that documents reference each other by ID in a similar way to relational databases.
+
+The following document represents a `User`:
+
+```json
+{ "_id": { "$oid": "60d5922d00581b8f0062e3a8" }, "name": "Ella" }
```

Comment:
This is how ObjectIDs are represented in the Client? DO we explain that somewhere else how to handle that?

Reply:
No, in the database. I will clarify.

Reply:
Changed to 'MongoDB document' to emphasize that it's not a client response

Reply:
Ah interesting - so equivalent to a "table" view for relational SQL.